### PR TITLE
Fix usage of PrintOptions

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Implement setting parameters of `FWCore.ParameterSet.Mixins.PrintOptions` in the constructor.

Clean up all calls of `dumpPython(...)` that were using the wrong parameter type (`None` or a string instead of a `PrintOptions` object).

Ths PR is a prerequisite for #27305 .

#### PR validation:

All unit tests pass.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
?

**Comment by fwyzard** [comment]
@cmsbuild, please abort

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
+1
Thanks Andrea

File: Validation/RecoEgamma/test/ElectronFromUncleanedOnly_cfg.py [hunk 1]
```diff
@@ -30,5 +30,4 @@
 #from Configuration.AlCa.autoCond import autoCond
 #process.GlobalTag.globaltag = autoCond[os.environ['TEST_GLOBAL_AUTOCOND']]
 
-process.dumpPython(None)
```
[the remark this answers is not in the document]
Comment (Sam-Harper) [reply Validation/RecoEgamma/test/ElectronFromUncleanedOnly_cfg.py]: it looks super old to me and couldnt do anything of use. If @lsoffi , @jainshilpi  dont object, I would just delete it.

[the remark this answers is not in the document]
Comment (fabiocos) [reply Validation/RecoEgamma/test/ElectronFromUncleanedOnly_cfg.py]: @Sam-Harper thanks

**Comment by fabiocos** [comment]
+operations

direct comparison of python output of ```ConfigBuilder.py``` in ```runTheMatrix.py``` shows no difference with the adoption of this PR

**Comment by fabiocos** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**